nilearn/plotting/img_plotting.py

```diff
@@ -760,10 +771,55 @@ def plot_glass_brain(stat_map_img,
 
     """
     if cmap is None:
-        cmap = plt.cm.hot if black_bg else plt.cm.hot_r
+        cmap = cm.cold_hot if black_bg else cm.cold_white_hot
+
+    if stat_map_img:
+        # make sure that the color range is symmetrical
+        if vmax is None or symmetric_cbar in ['auto', False]:
+            stat_map_img = _utils.check_niimg_3d(stat_map_img)
+            stat_map_data = stat_map_img.get_data()
+            # Avoid dealing with masked_array:
+            if hasattr(stat_map_data, '_mask'):
+                stat_map_data = np.asarray(stat_map_data[
+                        np.logical_not(stat_map_data._mask)])
+            stat_map_max = np.nanmax(stat_map_data)
+            if not plot_abs:
+                stat_map_min = np.nanmin(stat_map_data)
+            else:
+                stat_map_min = 0
+
+        if symmetric_cbar == 'auto':
+            symmetric_cbar = (stat_map_min < 0) and (stat_map_max > 0)
+
+        if vmax is None:
+            vmax = max(-stat_map_min, stat_map_max)
+
+        if 'vmin' in kwargs:
+            raise ValueError('plot_stat_map does not accept a "vmin" '
+                             'argument, as it uses a symmetrical range '
+                             'defined via the vmax argument. To threshold '
+                             'the map, use the "threshold" argument')
+        vmin = -vmax
+
+        if not symmetric_cbar:
+            negative_range = (stat_map_max <= 0)
+            positive_range = (stat_map_min >= 0)
+            if positive_range:
+                cbar_vmin = 0
+                cbar_vmax = None
+            elif negative_range:
+                cbar_vmax = 0
+                cbar_vmin = None
+            else:
+                cbar_vmin = stat_map_min
+                cbar_vmax = stat_map_max
+        else:
+            cbar_vmin, cbar_vmax = None, None
+    else:
+        cbar_vmin, cbar_vmax = None, None
```

Comment:
I guess this whole block doing the tricky is taken from plot_stat_map. Is there a simple way to factor that in a function ?